packages/foundry/src/create.ts

```diff
@@ -7,11 +7,21 @@ const BaseRepositoryAddress = "workleap/wl-foundry-cli/templates";
 
 type TemplateId = "host-application" | "remote-module" | "static-module";
 
+const validNpmPackageNameRegex = /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/;
+
+function validatePackageName(scope:string, name:string) {
+    if (!validNpmPackageNameRegex.test(`${scope}/${name}`)) {
+        throw new Error(`Invalid package.json name "${scope}/${name}"`);
+    }
+}
+
 const TemplateGenerators: Record<TemplateId, (outputDirectory: string, options: Record<string, string>) => Promise<void>> = {
     "host-application": async (outputDirectory, options) => {
         const scope = options["packageScope"];
         const name = basename(options["outDir"]);
 
+        validatePackageName(scope, name);
```

Comment:
I think its a bit late at this point to do validation, as the command is now running. 

Can we run the validNpmPackageNameRegex on the scope during the initial validation?

Reply:
We can validate parts of the name

Reply:
Given that we will ask for the package name is the client (create-workleap) the name could be validated at that time as we do for other arguments.

What do you think @alexasselin008 ?

Reply:
Yeah

Reply:
Perfect, I will oupdate the flow and move that validation

Reply:
Finally, the new option will be done in the next PR and the package name validation will also be added in this next PR.